keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/DoubleVertexTest.java

```diff
@@ -89,4 +91,11 @@ public void canConcat() {
         DoubleVertex concatDimOne = DoubleVertex.concat(1, A, B);
         assertArrayEquals(concatDimOne.getShape(), new long[]{2, 4});
     }
+
+    @Test
+    public void canCastToIntegerVertex() {
+        DoubleVertex v1 = new ConstantDoubleVertex(3.6);
+        IntegerVertex intV1 = v1.toInteger();
+        assertEquals(3, intV1.getValue().scalar(), 0.0);
```

Comment:
drop the 3rd parameter (delta=0.0). You want the `assertEquals(long, long)` version of this method, not `assertEquals(double, double, double)`.